### DISCUSSION TITLE
Iconlayer is not following pathlayer and icon is not displaying properly

### DISCUSSION THREAD

**Discussion by ats1999** [visgl/deck.gl#5266] [Q&A]
## 1. Icon layer is not following the path
> Code is in this  repository https://github.com/ats1999/rahul-react-map-gl-cluster/tree/main/react-map-gl-cluster
Only the `deck.gl` folder is related to this issue.

> Code is live on https://codesandbox.io/s/misty-grass-lbfgn?file=/src/App.js, but here icon not displaying 

> I only need one icon, so i used only one icon for the icon layer

_**see the position of the icon at different longitude and longitude along with the `PathLayer`**_
![Screenshot (504)](https://user-images.githubusercontent.com/54087826/102621906-662b2180-4194-11eb-9c35-fd8e5d5fa548.png)
![Screenshot (506)](https://user-images.githubusercontent.com/54087826/102621926-6aefd580-4194-11eb-808b-1ae86bb12800.png)
![Screenshot (507)](https://user-images.githubusercontent.com/54087826/102621947-7216e380-4194-11eb-96f2-d43344e81c13.png)
![Screenshot (508)](https://user-images.githubusercontent.com/54087826/102621965-793df180-4194-11eb-9739-9937858f89fb.png)

_**You can notice, it is taking some offset at different points**_
I passed this property to IconLayer
```js
getPixelOffset:[0,50]
// [0,0] is giving some offset to the icon, but it remains consistent along the path
```

## 2. IconLayer is not displaying properly 
https://codesandbox.io/s/misty-grass-lbfgn?file=/src/App.js

I have used the same SVG image in the above-mentioned repository, whose image is shown above. That is a long codebase and points are coming from the remote server, so I am not able to share it with you. But, I created the same situation that can describe the problem.

When you run the above repository after cloning.  you can notice, the icon is not displaying the same as it is shown in the above image. But, here also i have used the same image and same `IconLayer` property.

**Comment by Pessimistress** [comment]
Image loading is not synchronous. When you supply the image to `iconAtlas`, it is not yet loaded. If you do not know how to fix this, just supply the svg directly to `iconAtlas` as an URL.

Your sandbox shows the icon following the path nicely. You should check if the icon coordinate has the correct z value in your own app.

**Reply by ats1999** [reply]
Only png image is being displayed: https://res.cloudinary.com/bdevg/image/upload/v1606483976/right-arrow_pdxgkq.png
this SVG image is not being displayed: https://res.cloudinary.com/bdevg/image/upload/v1608352746/right-arrow_f8fn9u.svg

But, the same SVG is being displayed at the right side of `Hello, thanks for your time`.

**Reply by Pessimistress** [reply]
If you look in your console and search for the error in deck.gl issues you can find https://github.com/visgl/deck.gl/issues/5158
